ts/components/ui/ListItemTransaction.tsx

```diff
@@ -0,0 +1,194 @@
+import * as O from "fp-ts/lib/Option";
+import { pipe } from "fp-ts/lib/function";
+import * as React from "react";
+import { ImageURISource, StyleSheet, View } from "react-native";
+import Placeholder from "rn-placeholder";
+import I18n from "../../i18n";
+import { WithTestID } from "../../types/WithTestID";
+import { IOBadge } from "../core/IOBadge";
+import { Icon } from "../core/icons";
+import { IOLogoPaymentType, LogoPayment } from "../core/logos";
+import { VSpacer } from "../core/spacer/Spacer";
+import { LabelSmall } from "../core/typography/LabelSmall";
+import { NewH6 } from "../core/typography/NewH6";
+import { useIOTheme } from "../core/variables/IOColors";
+import {
+  IOListItemStyles,
+  IOListItemVisualParams,
+  IOStyles
+} from "../core/variables/IOStyles";
+import Avatar from "./Avatar";
+import {
+  PressableBaseProps,
+  PressableListItemBase
+} from "./utils/baseListItems";
+
+type LogoNameOrUri = IOLogoPaymentType | ImageURISource;
+export type ListItemTransaction = WithTestID<
+  PressableBaseProps & {
+    hasChevronRight?: boolean;
+    isLoading?: boolean;
+    leftPaymentLogoOrUrl?: LogoNameOrUri;
+    subtitle: string;
+    title: string;
+  } & (
+      | {
+          transactionStatus: "success";
+          transactionAmount: string;
+        }
+      | {
+          transactionStatus: "failure" | "pending";
+          transactionAmount?: string;
+        }
+    )
+>;
+
+type LeftComponentProps = {
+  logoNameOrUrl: LogoNameOrUri;
+};
+
+const isImageUrI = (
+  value: IOLogoPaymentType | ImageURISource
+): value is ImageURISource =>
+  typeof value === "object" && value.uri !== undefined;
+
+const LeftComponent = ({ logoNameOrUrl }: LeftComponentProps) => {
+  if (isImageUrI(logoNameOrUrl)) {
+    return <Avatar shape="circle" size="small" logoUri={[logoNameOrUrl]} />;
+  } else {
+    return <LogoPayment size={44} name={logoNameOrUrl} />;
```

Comment:
To avoid large payment logos, we should put them in a container with the same size as the `Avatar` component and leave the default size (`24`). Here's my suggestion:

```suggestion
    return (
      <View
        style={{
          width: IOVisualCostants.avatarSizeSmall,
          height: IOVisualCostants.avatarSizeSmall,
          alignItems: "center",
          justifyContent: "center"
        }}
      >
        <LogoPayment name={logoNameOrUrl} />
      </View>
    );
```

Reply:
The `LogoPayment` component already has a small default size. But we designed the component to accommodate both large and small assets, specifically the `Avatar` (which couldn't be smaller) and the smaller `LogoPayment`. The visual misalignment is solved by using the larger component to contain the smaller one.